zsshlib/oidc.go

```diff
@@ -0,0 +1,40 @@
+package zsshlib
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/oauth2"
+
+	"github.com/sirupsen/logrus"
+)
+
+func OIDCFlow(initialContext context.Context, flags SshFlags) (string, error) {
+	callbackPath := "/auth/callback"
+	cfg := &Config{
+		Config: oauth2.Config{
+			ClientID:     flags.OIDC.ClientID,
+			ClientSecret: flags.OIDC.ClientSecret,
+			RedirectURL:  fmt.Sprintf("http://localhost:%v%v", flags.OIDC.CallbackPort, callbackPath),
```

Comment:
It may be more intuitive for the user to configure the callback as an origin string w/ path, not discrete port and path, because that is how it will appear in the OAuth provider's configuration where it's authorized for this particular client ID, e.g. `http://localhost:1234/auth/callback`.

Reply:
not sure i follow but we can discuss in the future